pkg/operator/status/status_controller.go

```diff
@@ -0,0 +1,407 @@
+			return isWatchedSecret(platformType, e.Meta.GetNamespace(), e.Meta.GetName(), e.Meta.GetAnnotations())
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isWatchedSecret(platformType, e.Meta.GetNamespace(), e.Meta.GetName(), e.Meta.GetAnnotations())
+		},
+	}
+
+	// Whenever one of our watched Secrets is updated, recalculate status
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(alwaysReconcileCCOConfigObject),
+		},
+		p,
+	)
+
+	// TODO: Watch the Authentication object for changes as that can cause the OIDC controller to change status
+	// (but something as simple as being unable to Get() the Authentication can cause degraded and we will not
+	// be notified of that).
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileStatus{}
+
+// ReconcileStatus reconciles the entire status for this operator.
+type ReconcileStatus struct {
+	client.Client
+	Logger   log.FieldLogger
+	platform configv1.PlatformType
+}
+
+// Reconcile will ensure the ClusterOperator status conditions are updating by calling to each
+// registered domain-specific status handler. This function will ensure default conditions are set
+// if none of the handlers set a condition.
+func (r *ReconcileStatus) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	start := time.Now()
+
+	defer func() {
+		dur := time.Since(start)
+		metrics.MetricControllerReconcileTime.WithLabelValues(controllerName).Observe(dur.Seconds())
+	}()
+
+	err := SyncStatus(r.Client, r.Logger)
+	return reconcile.Result{
+		RequeueAfter: defaultRequeuePeriod,
+	}, err
+}
+
+// SyncStatus allows triggering a status sync from outside of this controller.
+func SyncStatus(kubeClient client.Client, logger log.FieldLogger) error {
+	log.Info("reconciling clusteroperator status")
+
+	co := &configv1.ClusterOperator{}
+	err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: constants.CloudCredClusterOperatorName}, co)
+	isNotFound := k8errors.IsNotFound(err)
+	if err != nil && !isNotFound {
+		logger.WithError(err).WithField("clusterOperator", constants.CloudCredClusterOperatorName).Error("failed to retrive ClusterOperator")
+		return err
+	}
+
+	oldConditions := co.Status.Conditions
+	oldVersions := co.Status.Versions
+	oldRelatedObjects := co.Status.RelatedObjects
+
+	// We rebuild the conditions from scratch each time.
+	// Handlers return abnormal (non-default) conditions they wish to set.
+	// if the controller is functioning normally, it should return an empty slice of conditions.
+	conditions := []configv1.ClusterOperatorStatusCondition{}
+	relatedObjects := []configv1.ObjectReference{}
+	for handlerName, handler := range statusHandlers {
+		handlerConditions, err := handler.GetConditions(logger)
+		if err != nil {
+			logger.WithError(err).WithField("statushandler", handlerName).Errorf("failed to get conditions from status handler")
+			continue
+		}
```

Comment:
Could you info log each condition we got here and the status handler it came from. Then below when we see a conflict we can log that as a warning and still know the user/admin can see which handlers reported the same condition.